### ISSUE TITLE
Seems I need Rosettadrone for a science project again. 

### ISSUE THREAD

**Issue by The1only** [The1only/rosettadrone#124]
What is the general status these days?
I will use the Mavic 2 Pro with a smartcontroller, Dronekit, and python API, and a NVIDIA Orin GPU. 
I plan to compile the system from a MAC M1.
Any heads up?

**Comment by kripper** [comment]
Hi Terje,

Please use master and always test with the simulator the first time after fetching new changes (just in case).

When flying a real drone, as you probably know, just disconnect the RC if you notice something is going wrong.

All your old features are supposed to be there even when a lot of code has been (and is being) rewritten.

Your custom code has been temporarily moved to the PluginManager class. You should later move it into a Plugin class and add additional "hooks" in MainActivity if necessary.

If you experience any problem with latest changes, I'll be here to help you.

Rosetta now supports flying in QGC without actually connecting a drone. Just enter TestMode.

I'm currently playing with Gazebo to simulate video input.

**Comment by The1only** [comment]
Perfect, the code compiled right out of the box. 

Did you ever find out about the Gradle issue?  ( We must use an old Gradle to be able to get video?)

**Comment by kripper** [comment]
I'm not aware of this issue.
In case of any video issue, please test this sample app fork:
https://github.com/kripper/DJI-Android-VideoStreamDecodingSample

This code should replace all video code in Rosetta once tested with many models.

**Comment by The1only** [comment]
So if you in AndroidStudio accept all suggested updates, the app still works now?

**Comment by The1only** [comment]
I added all updates and now can't compile. Ever seen this:
(in materials-1.8.0/res/values)

Can't determine type for tag '_<macro name_=_"m3_comp_assist_chip_container_shape"_>_?attr/shapeAppearanceCornerSmall</macro>'

**Comment by kripper** [comment]
Gradle updates? No, I haven't tested updating Gradle and have never seen this error.

**Comment by The1only** [comment]
OK, not important, just tested.

**Comment by The1only** [comment]
PS: the DJI SDK can be updated from Android 4.16.1 to Android 4.16.4
It requires one small fix only in Rosettadrone, the fix from DJI is: 

The US RID function does not work properly on the DJI RC Pro remote controller and some mobile phones, resulting in the aircraft unable to take off: Fixed.

**Comment by The1only** [comment]
Something is strange, are you sure the master is working? 
I get a lot of Mavlink errors, and the app can not run, even if I compile fresh after git clone ???

**Comment by The1only** [comment]
Did anyone use a DJI smart-controller lately ??? It can no longer run Rosettadrone somehow. 
At least on my built.

**Comment by kripper** [comment]
Sorry, I don't have a smart controller to test.
I remember somebody posted an issue regarding the smart controller time ago.
Whatever problem you are experiencing, it's probably easy to fix if it worked once and broke somehow.

**Comment by kripper** [comment]
https://github.com/The1only/rosettadrone/issues/105

**Comment by kripper** [comment]
What errors are you seeing?

**Comment by The1only** [comment]
I got it working, it's a balancing act on the dependencies. I spent some hours trying to find what needed updating and to what level. I did changes to app/build.gradle, a small change in dummyProduct.java ( to support DJI std 14.16.4 ), and a tiny fix in Manifest (not mandatory but more best practice) :-)
Seems it all works now, let me know if you want me to create a pull request. You are the boss now !
